core/include/kernel/mutex.h

```diff
@@ -76,6 +76,27 @@ void mutex_unlock_recursive(struct recursive_mutex *m);
 void mutex_lock_recursive(struct recursive_mutex *m);
 #endif
 
+/*
+ * struct mutex_pm_aware - Mutex usable in PM atomic sequence
+ *
+ * Some resources need a mutex protection for runtime operations but are
+ * also accessed during specific system power transition (PM power off,
+ * suspend and resume) that operate in atomic execution environment where
+ * non-secure world is not operational, for example in fastcall SMC entries
+ * of the PSCI services. In such case we cannot take a mutex and we expect
+ * the mutex is unlocked. Additionally a spinning lock is attempted to be
+ * locked to check the resource access consistency.
+ */
+struct mutex_pm_aware {
+	struct mutex mutex;	/* access protection in thread context */
+	unsigned int lock;	/* access consistency in PM context */
+};
+
```

Comment:
Wouldn't a `MUTEX_PM_AWARE_INITIALIZER` make sense too?

Reply:
Sorry, it's not for that reason. But the conflict is there. I'll try to find a way to define it and will tell you.